### ISSUE TITLE
Apparent different docker container create from plugin vs Dockerfile

### ISSUE THREAD

**Issue by mltalley** [spotify/docker-maven-plugin#281]
When I create a docker image from the plugin I get an error when trying to run the docker image.  However, when I run the same image using /bin/bash as the entrypoint and then execute the same entrypoint command it runs just fine.  Also if I use a Dockerfile to create the same image, it works just fine doing a docker run.
## How to reproduce

 **POM file configuration:**

```
            <plugin>
              <groupId>com.spotify</groupId>
              <artifactId>docker-maven-plugin</artifactId>
              <version>0.4.13</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>build</goal>
                        </goals>
                        <phase>package</phase>
                    </execution>
                </executions>
              <configuration>
                <imageName>myImage</imageName>
                <baseImage>local.docker.registry:5000/openjdk-7:latest</baseImage>
                <entryPoint>["java", "-Dspring.profiles.active=local -DCCP_CONFIG_FILE=configCCP.xml -jar", "/${project.build.finalName}.jar"]</entryPoint>
                <!-- copy the service's jar file from target into the root directory of the image --> 
                <resources>
                   <resource>
                     <targetPath>/</targetPath>
                     <directory>${project.build.directory}</directory>
                     <include>${project.build.finalName}.jar</include>
                   </resource>
                </resources>
              </configuration>
            </plugin>
```

**Output from build:**

```
[INFO] --- docker-maven-plugin:0.4.13:build (default) @ MYService ---
[INFO] Copying /home/user/git/project/MyService/target/MyService-1.1.0-SNAPSHOT.jar -> /home/user/git/project/MyService/target/docker/MyService-1.1.0-SNAPSHOT.jar
[INFO] Building image myImage
Step 1 : FROM local.docker.registry:5000/openjdk-7:latest
 ---> 63ec3c344602
Step 2 : ADD /MyService-1.1.0-SNAPSHOT.jar //
 ---> 4c8ae9776dce
Removing intermediate container 34e0431d7254
Step 3 : ENTRYPOINT java -Dspring.profiles.active=local -DCCP_CONFIG_FILE=configCCP.xml -jar /MyService-1.1.0-SNAPSHOT.jar
 ---> Running in b053fa169cf6
 ---> 9e9a6f47f969
Removing intermediate container b053fa169cf6
Successfully built 9e9a6f47f969
[INFO] Built myImage
```

**Output from run command:**

```
docker run 9e9a6f47f969
Error: Could not find or load main class .MyService-1.1.0-SNAPSHOT.jar
```

If I use this command to get a bash terminal in the container I can see the MyService jar file, and the entrypoint script works fine: docker run -ti --entrypoint=/bin/bash 9e9a6f47f969

**Also, when I use this Dockerfile to make an image, it will execute the entrypoint script without problems:**

```
FROM local.docker.registry:5000/openjdk-7
# get the latest version myService
ADD myService/target/myService-1.1.0-SNAPSHOT.jar /
# run it
ENTRYPOINT java -Dspring.profiles.active=local -DCCP_CONFIG_FILE=configCCP.xml -jar /myService-1.1.0-SNAPSHOT.jar

Step 1 : FROM local.docker.registry:5000/openjdk-7
 ---> 63ec3c344602
Step 2 : ADD myService/target/myService-1.1.0-SNAPSHOT.jar /
 ---> dcecb029f8d8
Removing intermediate container 417e4dd3348b
Step 3 : ENTRYPOINT java -Dspring.profiles.active=local -DCCP_CONFIG_FILE=configCCP.xml -jar /myService-1.1.0-SNAPSHOT.jar
 ---> Running in 0b18295da8ab
 ---> 5f802dbdff29
Removing intermediate container 0b18295da8ab
Successfully built 5f802dbdff29
```
## What do you expect

Both containers run the entrypoint command.
## What happened instead

Only the image that I build using the dockerfile will run the entrypoint command without error.
## Software:
- docker version: Docker version 1.9.1, build a34a1d5
- docker-maven-plugin version: 0.4.13
- maven version: Apache Maven 3.2.5 (12a6b3acb947671f09b81f49094c53f426d8cea1; 2014-12-14T10:29:23-07:00)

**Comment by mltalley** [comment]
It appears that the plug-in is very finicky about how you specify the entrypoint.  I was doing it this way:
`<entryPoint>["java", "-Dspring.profiles.active=local -DCCP_CONFIG_FILE=configCCP.xml -jar", "/${project.build.finalName}.jar"]</entryPoint>`
Notice that all the -D args and the -jar params are in the same quote block.

But when I did a docker inspect on both the container the plug-in created and the one I created with a docker build and a dockerfile I noticed the EntryPoint: was different:
**plug-in generated image:**
`"Entrypoint": [
            "java",
            "-Dspring.profiles.active=local -DCCP_CONFIG_FILE=configCCP.xml -jar",
            "/myService-1.1.0-SNAPSHOT.jar"
        ],
`
**docker build created image:**
`"Entrypoint": [
            "/bin/sh",
            "-c",
            "java -Dspring.profiles.active=local -DCCP_CONFIG_FILE=configCCP.xml -jar /myService-1.1.0-SNAPSHOT.jar"
        ],
`

So I tried changing how I specified the entrypoint in the maven pom file, and finally noticed that if I quoted everything separately it'd create the entrypoint differently:
**New pom file**
`<entryPoint>["java", "-Dspring.profiles.active=local",  "-DCCP_CONFIG_FILE=configCCP.xml",  "-jar", "/${project.build.finalName}.jar"]</entryPoint>`

**Now the plug-in entrypoint looks like this (from inspect again):**
` "Entrypoint": [
            "java",
            "-Dspring.profiles.active=local",
            "-DCCP_CONFIG_FILE=configCCP.xml",
            "-jar",
            "/myService-1.1.0-SNAPSHOT.jar"
        ],
`
This works, and starts my springboot jar file when I do a docker run.  However, it's still not the same as the entrypoint that doing a docker build creates.  

Anyone know why that may be?  And is it important to have that /bin/sh -c ??
